catroid/src/org/catrobat/catroid/ui/dialogs/OrientationDialog.java

```diff
@@ -143,4 +125,53 @@ public void setCreateEmptyProject(boolean isChecked) {
 	public void setCreateDroneProject(boolean isChecked) {
 		createDroneProject = isChecked;
 	}
+
+	private class LoadingOnCreatingProject extends AsyncTask<String, Void, Boolean> {
+		private ProgressDialog dialog;
+
+		protected LoadingOnCreatingProject(Context context) {
+			dialog = new ProgressDialog(context);
+		}
+
+		@Override
+		protected void onPostExecute(Boolean showing) {
+			super.onPostExecute(showing);
+			dialog.dismiss();
+		}
+
+		@Override
+		protected void onPreExecute() {
+			super.onPreExecute();
+			this.dialog.setMessage(getResources().getString(R.string.please_wait));
+			this.dialog.show();
+		}
+
+		@Override
+		protected Boolean doInBackground(String... params) {
+			try {
+				try {
+					ProjectManager.getInstance().initializeNewProject(projectName, getActivity(), createEmptyProject, createDroneProject, createLandscapeProject);
+				} catch (IllegalArgumentException illegalArgumentException) {
+					Utils.showErrorDialog(getActivity(), R.string.error_project_exists);
+				} catch (IOException ioException) {
+					Utils.showErrorDialog(getActivity(), R.string.error_new_project);
+					Log.e(TAG, Log.getStackTraceString(ioException));
+					dismiss();
+				}
```

Comment:
The original code had a return in every catch block. You need it as well to stop at these points.